### PR TITLE
core: fix nonce verification one more time

### PR DESCRIPTION
**Pull request by fjl**
The block nonce verification was effectively disabled by a typo (missing `!`).
This time, there is an actual test for it.

The bug was introduced in e7e2cbfc018f5e0. This commit is on the develop branch but has not appeared in any release yet. It was reviewed by two people before inclusion. This bug has lead to a fork on the Olympic test network because some people run nodes with code from the develop branch.

### CODE REVIEW

File: core/chain_manager.go [hunk 1]
```diff
@@ -551,12 +551,12 @@ func (self *ChainManager) InsertChain(chain types.Blocks) (int, error) {
 		bstart := time.Now()
 		// Wait for block i's nonce to be verified before processing
 		// its state transition.
-		for nonceChecked[i] {
+		for !nonceChecked[i] {
```
Comment (zelig) [inline core/chain_manager.go]: ouch

File: core/chain_manager_test.go [hunk 2]
```diff
@@ -426,3 +427,55 @@ func TestReorgShortest(t *testing.T) {
...
+		}
+		nerr := err.(*BlockNonceErr)
+		if nerr.Number.Cmp(failblock.Number()) != 0 {
+			t.Errorf("(i=%d) wrong block number in error, got %v, want %v", i, nerr.Number, failblock.Number())
+		}
+		if nerr.Hash != failblock.Hash() {
+			t.Errorf("(i=%d) wrong block hash in error, got %v, want %v", i, nerr.Hash, failblock.Hash())
+		}
+
+		// Check that all no blocks after the failing block have been inserted.
```
Comment (ethers) [inline core/chain_manager_test.go]: minor: clearer if "all" is removed ("all no" is not correct grammar).

File: core/chain_manager.go (hunk 1 above)
Comment (ABISprotocol) [inline core/chain_manager.go]: wow

File: core/chain_manager_test.go (hunk 2 above)
Comment (fjl) [inline core/chain_manager_test.go]: Good find. Thank you.

**Pull request merged**